mem-test/tests/dhat_large_table_log.rs

```diff
@@ -0,0 +1,138 @@
+use tracing::info;
+use url::Url;
+
+#[global_allocator]
+static ALLOC: dhat::Alloc = dhat::Alloc;
+
+const NUM_COMMITS: u64 = 1_000; // number of commit files to generate
+const TOTAL_ACTIONS: u64 = 1_000_000; // total add/remove actions across all commits
+
+/// Generate a delta log with NUM_COMMITS commit files and TOTAL_ACTIONS add/remove action pairs
+fn generate_delta_log(path: &Path) -> Result<(), Box<dyn std::error::Error>> {
+    let delta_log_path = path.join("_delta_log");
+    create_dir_all(&delta_log_path)?;
+
+    let actions_per_commit = TOTAL_ACTIONS / NUM_COMMITS;
+    let mut current_file_id = 0u64;
+
+    for commit_id in 0..NUM_COMMITS {
+        let commit_filename = format!("{:020}.json", commit_id);
+        let commit_path = delta_log_path.join(&commit_filename);
+        let mut file = File::create(&commit_path)?;
+
+        let mut actions = Vec::new();
+
+        // Add metadata and protocol only for the first commit
+        if commit_id == 0 {
+            actions.push(json!({
+                "metaData": {
+                    "id": format!("{:020}", commit_id),
+                    "format": { "provider": "parquet", "options": {} },
+                    "schemaString": r#"{"type":"struct","fields":[{"name":"value","type":"integer","nullable":true,"metadata":{}}]}"#,
+                    "partitionColumns": [],
+                    "configuration": {}
+                }
+            }));
+            actions.push(json!({
+                "protocol": {
+                    "minReaderVersion": 1,
+                    "minWriterVersion": 1
+                }
+            }));
+        }
+
+        // Generate add/remove action pairs for this commit
+        for _ in 0..actions_per_commit / 2 {
+            // Add action
+            actions.push(json!({
+                "add": {
+                    "path": format!("{}.parquet", current_file_id),
+                    "partitionValues": {},
+                    "size": 1000000,
+                    "modificationTime": commit_id * 1000,
+                    "dataChange": true
+                }
+            }));
+
+            // Remove action (remove a previous file if we have any)
+            if current_file_id > 0 {
+                actions.push(json!({
+                    "remove": {
+                        "path": format!("{}.parquet", current_file_id - 1),
+                        "deletionTimestamp": commit_id * 1000 + 500,
+                        "dataChange": true
+                    }
+                }));
+            }
+
+            current_file_id += 1;
+        }
+
+        // Write actions to file
+        for action in actions {
+            writeln!(file, "{}", action)?;
+        }
+    }
+
+    info!(
+        "Generated {} commit files with {} total actions",
+        NUM_COMMITS, TOTAL_ACTIONS
+    );
+    Ok(())
+}
+
+#[ignore = "mem-test - run manually"]
+#[test]
+fn test_dhat_large_table_log() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
oof 
```
Failed to get latest snapshot: ObjectStore(Generic { store: "LocalFileSystem", source: UnableToOpenFile { source: Os { code: 24, kind: Uncategorized, message: "Too many open files" }, path: "/var/folders/cr/cl0ybfwx6kjd8bybfc2mr4rw0000gp/T/.tmpapqcBg/_delta_log/00000000000000000749.json" } })
```

Reply:
ah macos? i think might need to bump up ulimit